### PR TITLE
added currency converter plugin

### PR DESCRIPTION
**Pull request by jitunayak**
Screenshot 2021-03-05 at 1 05 16 PM

I added the converter plugin for currency conversion 🛠 

e.g

100 INR in USD
99 usd in eur
9 USD into INR

### CODE REVIEW

File: src/plugins/converter/converter.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { SpotterOption, SpotterPlugin, SpotterPluginLifecycle } from '../../core';
+import icon from './icon.png';
+export class ConverterPlugin extends SpotterPlugin implements SpotterPluginLifecycle {
+
+  identifier = 'Converter';
+
+  async onQuery(query: string): Promise<SpotterOption[]> {
+       
+    try {
+       const splitted = query.split(" ",4)
```
Comment (ziulev) [inline src/plugins/converter/converter.plugin.ts:10]: `1euro in USD` without space won't work. I think we need a regexp here or so on.

File: src/plugins/converter/converter.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+       const convertFromAmount = Number(splitted[0])
+       const convertFromCurrencyType = splitted[1].toUpperCase()
+       const convertToCurrencyType = splitted[3].toUpperCase()
+       
+       if(splitted.length==4 && convertToCurrencyType.length==3)
+       {
+        var result=0
```
Comment (ziulev) [inline src/plugins/converter/converter.plugin.ts:17]: Remove unused variable please. And use `let` or `const` to declare variables

File: src/plugins/converter/converter.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        const data = await fetch(`https://api.exchangeratesapi.io/latest?base=${convertFromCurrencyType}&symbols=${convertToCurrencyType}`)
+        const json = data.json()
+        const r = json.then(data =>{
```
Comment (ziulev) [inline src/plugins/converter/converter.plugin.ts:20]: Please use human understandable variables 🙂.

But here could be match simple
```ts
const data = await fetch(`https://api.exchangeratesapi.io/latest?base=${convertFromCurrencyType}&symbols=${convertToCurrencyType}`).then(d => d.json());
const rate = ...
```

**Review by ziulev — CHANGES_REQUESTED** [review]
Awesome! Thank you for your job! 🚀

Please check all comments, and make sure your linter doesn't have any warnings.
There are few places without `;`, unusable variables, etc

**Comment by relikd** [comment]
Haven't looked at the plugin maybe its already implemented, but, it would be nice to input a single price (e.g., $100) and see the output of multiple. Not that I am interested in multiple currency but because I am too lazy to type "100 usd to eur". Often I need an approximate conversion of a foreign currency into local currency. But since there are no preferences for the local currency, just display the few most common ones (similar to spotlight which always shows 6 currencies).

File: src/plugins/converter/converter.plugin.ts (hunk 3 above)
Comment (relikd) [reply src/plugins/converter/converter.plugin.ts:20]: also, cache and reuse results. No need to query the server mutltiple times when a single request to download and store the conversion rate is enough.

**Pull request closed without merging**